src/queue/multilevel.rs

```diff
@@ -533,43 +554,48 @@ impl TaskElapsedMap {
                 v.clone()
             }
         };
-        TLS_LAST_CLEANUP_TIME.with(|t| {
-            fail_point!("cleanup-in-get-elapsed", |_| ());
-            if recent().saturating_duration_since(t.get()) > self.cleanup_interval {
-                self.maybe_cleanup();
-            }
-        });
+        self.maybe_cleanup();
         elapsed
     }
 
+    fn try_cleanup(&self) -> Option<Instant> {
+        self.cleaning_up
+            .compare_exchange(false, true, SeqCst, SeqCst)
+            .map(|_| {
+                let old_index = self.new_index.load(SeqCst) ^ 1;
+                self.maps[old_index].clear();
```

Comment:
`clear` acquires locks inside the map, so I'm thinking if we want to avoid tail latency completely, maybe it's better `swap` a new map in.

Reply:
We can't avoid the cost by swapping a new map because the cost of Drop(map) is almost the same as  map.clear.

Reply:
I see a lot of tasks spawned by tikv side just use a random task id, e.g. point_get tasks, so maybe we can avoid putting a track entry for these tasks as there are totally useless.

Reply:
> We can't avoid the cost by swapping a new map because the cost of Drop(map) is almost the same as map.clear, drop each entry(Arc::drop) is the main cost.

I think the point of this PR is to move map cleaning out of the `spawn` routine. For example, `try_cleanup` can be called at intervals in an external background thread.

Reply:
> I see a lot of tasks spawned by tikv side just use a random task id, e.g. point_get tasks, so maybe we can avoid putting a track entry for these tasks as there are totally useless.

This is a good point. Yet I think this PR is still valuable considering there are also many cases of explicit transactions.

Reply:
I'm not against this PR, I think it can help reduce the maximum latency a bit. I just want to point out that the cost of clean up can't be totally avoided.

Reply:
> I'm not against this PR, I think it can help reduce the maximum latency a bit. I just want to point out that the cost of clean up can't be totally avoided.

The amortized cost of clean up is tiny. So, actually we don't care about the cost. Maximum latency is the only problem we want to solve here.

If we can put in a new map and mark the old one as stale, it's possible for us to avoid any lock contension with clean up in `get_elapsed`.

Reply:
I found we can just change the current `old_map.get(&key)` to `old_map.try_get(&key)` to avoid this lock contension so if the old_map is still under cleaning, we just insert a new entry to the new_map instead. Because if the old_map is under cleaning(wlock), waiting for it is meanless because we are sure to get None after the wlock is released.

Reply:
Oh, I didn't know that. It's definitely better than swapping.